### ISSUE TITLE
Gas price unavailable

### ISSUE THREAD

**Issue by ronaldt80** [fsaris/home-assistant-zonneplan-one#37]
Thanks for this integration. I received all data except for the gas price - integration throws below error (don't have solar panels with zonneplan and did not connect the logger so not too worried about the pv and P1 error). Anyone know how to address this?

```
This error originated from a custom integration.

Logger: custom_components.zonneplan_one.coordinator
Source: custom_components/zonneplan_one/coordinator.py:120
Integration: Zonneplan ONE (documentation, issues)
First occurred: 1 January 2023 at 15:53:37 (885 occurrences)
Last logged: 09:30:40

Could not find pv_installation of pv_installation
Could not find p1_installation of p1_installation
Could not find measurement_groups of gas_data.measurement_groups.0.meta.price
```

**Comment by YvoTuk** [comment]
Same here, when looking at the integration, the sensor Current Zonneplan Gas tariff doesn't show at all (not as disabled either)

**Comment by fsaris** [comment]
Can you enable debug and post the logs here? Could be that the api was changed. I don't have energy contract so I don't get this data

**Comment by YvoTuk** [comment]
Sorry, just found out how to do it.
I looked at the devices where I didn't see the entity, however, when you go to integrations and click on the entities, you will see (and are able to enable) all of them!

**Comment by ronaldt80** [comment]
So that's not the same. I have it enabled and mine say "unknown".

I enabled debug logging in the integration but no clue where to find the log file for an individual integration. 

All I see in the log is
`Could not find measurement_groups of gas_data.measurement_groups.0.meta.price`

**Comment by fsaris** [comment]
@ronaldt80 If you check the full log (below the logs you already found) it should show you the debug output.

Is your energy contract already active?

**Comment by ronaldt80** [comment]
The contract is active since last night; I can also see the gas price in the app. 

Familiar with the log you refer to but no debug output from the integration - I activated it in the integration tab but perhaps my yaml logger setting suppresses it. Will change logger settings in yaml and await the API refresh at 6 am coming morning. Perhaps it will resolve itself.

**Comment by Inferi0r** [comment]
Same for me, energy prices available (also in app), but gas prices 'Unknown' (not available in app yet either). Probably have to do that I am a new customer (contract starting at January 13).

**Comment by tieskuh** [comment]
> Same for me, energy prices available (also in app), but gas prices 'Unknown' (not available in app yet either). Probably have to do that I am a new customer (contract starting at January 13).

I can confirm gas prices are not visible until the contract starts.

**Comment by fsaris** [comment]
> I can confirm gas prices are not visible until the contract starts.

**Comment by ronaldt80** [comment]
My issue remains, contract has started on Jan 1, I can see the gas price in the app but not in the integration.  Would be curious to hear if on 13 Jan @Inferi0r can see the prices. Suggest to leave open until then.

**Comment by fsaris** [comment]
@ronaldt80 Did you restart HA? Please provide some logs

**Comment by ronaldt80** [comment]
Yep, restarted multiple times but unfortunately no change in behaviour. I provided logs already - continues to give the same:

`Could not find measurement_groups of gas_data.measurement_groups.0.meta.price`

Debug log for some reason doesn't work. Don't know why - switched it on in the integration tab (little bug icon is active) but nothing in the logs. Will check tonight if/how to activate this, it's not working for other integrations either so must be a global setting somewhere I overlooked.

**Comment by fsaris** [comment]
@ronaldt80 check https://github.com/fsaris/home-assistant-zonneplan-one#troubleshooting

**Comment by ronaldt80** [comment]
Thanks; followed those instructions and this worked!

Redacted the output a bit as everything (Bank details, address, full name etc) is in the api response 😄 

As you can see the gas price is in the data set at the 5am UTC time slot [logical I guess as this is 6 AM CET?] and is subsequently not converted / picked up in the sensor. By the way, is it normal that the integration generates 2 identical devices? I disabled one

`{'price': 2218777, 'electricity_price': 2218777, '**gas_price': 14781560**, 'tariff_group': 'low', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T05:00:00.000000Z', 'sustainability_score': 606, 'carbon_footprint_in_grams': 1747}`

`2023-01-04 21:13:17.660 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan current gas tariff: None [None]
`

**Detailed - redacted & compressed - log:**
```
2023-01-04 21:13:17.415 INFO (MainThread) [custom_components.zonneplan_one.api] fetch: /summary
2023-01-04 21:13:17.545 DEBUG (MainThread) [custom_components.zonneplan_one.api] ZonneplanAPI response header: <CIMultiDictProxy('Date': 'Wed, 04 Jan 2023 20:13:17 GMT', 'Server': 'Apache', 'Cache-Control': 'no-cache, private', 'X-RateLimit-Limit': '120', 'X-RateLimit-Remaining': '117', 'Vary': 'Authorization,User-Agent', 'Etag': '"8076935994a01d520ebf6a2ec5327470"', 'Transfer-Encoding': 'chunked', 'Content-Type': 'application/json')>
2023-01-04 21:13:17.545 DEBUG (MainThread) [custom_components.zonneplan_one.api] ZonneplanAPI response status: 200
2023-01-04 21:13:17.545 DEBUG (MainThread) [custom_components.zonneplan_one.api] ZonneplanAPI response body  : {'data': {'usage': {'value': 0, 'measured_at': None, 'type': 'normal', 'speed': 0, 'sustainability_score': 537, 'status_message': '', 'status_tip': 'Gemiddeld tarief'}, 'price_per_hour': [{'price': 3135110, 'electricity_price': 3135110, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-03T20:00:00.000000Z', 'sustainability_score': 435, 'carbon_footprint_in_grams': 3087}, {'price': 2938969, 'electricity_price': 2938969, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-03T21:00:00.000000Z', 'sustainability_score': 466, 'carbon_footprint_in_grams': 2893}, {'price': 2673495, 'electricity_price': 2673495, 'tariff_group': 'low', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-03T22:00:00.000000Z', 'sustainability_score': 508, 'carbon_footprint_in_grams': 2693}, {'price': 2523333, 'electricity_price': 2523333, 'tariff_group': 'low', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-03T23:00:00.000000Z', 'sustainability_score': 570, 'carbon_footprint_in_grams': 2562}, {'price': 2382611, 'electricity_price': 2382611, 'tariff_group': 'low', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T00:00:00.000000Z', 'sustainability_score': 619, 'carbon_footprint_in_grams': 2379}, {'price': 2013319, 'electricity_price': 2013319, 'tariff_group': 'low', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T01:00:00.000000Z', 'sustainability_score': 649, 'carbon_footprint_in_grams': 2114}, {'price': 1554487, 'electricity_price': 1554487, 'tariff_group': 'low', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T02:00:00.000000Z', 'sustainability_score': 651, 'carbon_footprint_in_grams': 1668}, {'price': 1554487, 'electricity_price': 1554487, 'tariff_group': 'low', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T03:00:00.000000Z', 'sustainability_score': 657, 'carbon_footprint_in_grams': 1633}, {'price': 1631080, 'electricity_price': 1631080, 'tariff_group': 'low', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T04:00:00.000000Z', 'sustainability_score': 645, 'carbon_footprint_in_grams': 1609}, {'price': 2218777, 'electricity_price': 2218777, '**gas_price': 14781560**, 'tariff_group': 'low', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T05:00:00.000000Z', 'sustainability_score': 606, 'carbon_footprint_in_grams': 1747}, {'price': 2586858, 'electricity_price': 2586858, 'tariff_group': 'low', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T06:00:00.000000Z', 'sustainability_score': 545, 'carbon_footprint_in_grams': 1955}, {'price': 3369487, 'electricity_price': 3369487, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T07:00:00.000000Z', 'sustainability_score': 501, 'carbon_footprint_in_grams': 2183}, {'price': 3321087, 'electricity_price': 3321087, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T08:00:00.000000Z', 'sustainability_score': 485, 'carbon_footprint_in_grams': 2082}, {'price': 3141886, 'electricity_price': 3141886, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T09:00:00.000000Z', 'sustainability_score': 517, 'carbon_footprint_in_grams': 2028}, {'price': 3133900, 'electricity_price': 3133900, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T10:00:00.000000Z', 'sustainability_score': 564, 'carbon_footprint_in_grams': 1999}, {'price': 3101714, 'electricity_price': 3101714, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T11:00:00.000000Z', 'sustainability_score': 599, 'carbon_footprint_in_grams': 2056}, {'price': 2995355, 'electricity_price': 2995355, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T12:00:00.000000Z', 'sustainability_score': 568, 'carbon_footprint_in_grams': 2254}, {'price': 2964742, 'electricity_price': 2964742, 'tariff_group': 'low', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T13:00:00.000000Z', 'sustainability_score': 560, 'carbon_footprint_in_grams': 2563}, {'price': 3103166, 'electricity_price': 3103166, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T14:00:00.000000Z', 'sustainability_score': 530, 'carbon_footprint_in_grams': 2631}, {'price': 3142612, 'electricity_price': 3142612, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T15:00:00.000000Z', 'sustainability_score': 500, 'carbon_footprint_in_grams': 2611}, {'price': 3371907, 'electricity_price': 3371907, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T16:00:00.000000Z', 'sustainability_score': 472, 'carbon_footprint_in_grams': 2704}, {'price': 3218237, 'electricity_price': 3218237, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T17:00:00.000000Z', 'sustainability_score': 464, 'carbon_footprint_in_grams': 2685}, {'price': 3121316, 'electricity_price': 3121316, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T18:00:00.000000Z', 'sustainability_score': 493, 'carbon_footprint_in_grams': 2674}, {'price': 3035527, 'electricity_price': 3035527, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T19:00:00.000000Z', 'sustainability_score': 513, 'carbon_footprint_in_grams': 2592}, {'price': 3120105, 'electricity_price': 3120105, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T20:00:00.000000Z', 'sustainability_score': 537, 'carbon_footprint_in_grams': 2435}, {'price': 2872902, 'electricity_price': 2872902, 'tariff_group': 'low', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T21:00:00.000000Z', 'sustainability_score': 564, 'carbon_footprint_in_grams': 2146}, {'price': 2837087, 'electricity_price': 2837087, 'tariff_group': 'low', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T22:00:00.000000Z', 'sustainability_score': 582, 'carbon_footprint_in_grams': 1783}, {'price': 1619827, 'electricity_price': 1619827, 'tariff_group': 'low', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-04T23:00:00.000000Z', 'sustainability_score': 613, 'carbon_footprint_in_grams': 1633}, {'price': 1598410, 'electricity_price': 1598410, 'tariff_group': 'low', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T00:00:00.000000Z', 'sustainability_score': 638, 'carbon_footprint_in_grams': 1473}, {'price': 1613777, 'electricity_price': 1613777, 'tariff_group': 'low', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T01:00:00.000000Z', 'sustainability_score': 630, 'carbon_footprint_in_grams': 1512}, {'price': 1554487, 'electricity_price': 1554487, 'tariff_group': 'low', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T02:00:00.000000Z', 'sustainability_score': 625, 'carbon_footprint_in_grams': 1543}, {'price': 1637614, 'electricity_price': 1637614, 'tariff_group': 'low', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T03:00:00.000000Z', 'sustainability_score': 618, 'carbon_footprint_in_grams': 1483}, {'price': 1751959, 'electricity_price': 1751959, 'tariff_group': 'low', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T04:00:00.000000Z', 'sustainability_score': 585, 'carbon_footprint_in_grams': 1416}, {'price': 2374867, 'electricity_price': 2374867, 'tariff_group': 'low', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T05:00:00.000000Z', 'sustainability_score': 528, 'carbon_footprint_in_grams': 1584}, {'price': 3005277, 'electricity_price': 3005277, 'tariff_group': 'low', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T06:00:00.000000Z', 'sustainability_score': 446, 'carbon_footprint_in_grams': 1530}, {'price': 3720387, 'electricity_price': 3720387, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T07:00:00.000000Z', 'sustainability_score': 400, 'carbon_footprint_in_grams': 1812}, {'price': 3624071, 'electricity_price': 3624071, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T08:00:00.000000Z', 'sustainability_score': 358, 'carbon_footprint_in_grams': 1959}, {'price': 3605074, 'electricity_price': 3605074, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T09:00:00.000000Z', 'sustainability_score': 349, 'carbon_footprint_in_grams': 2301}, {'price': 3417887, 'electricity_price': 3417887, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T10:00:00.000000Z', 'sustainability_score': 361, 'carbon_footprint_in_grams': 2255}, {'price': 3321087, 'electricity_price': 3321087, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T11:00:00.000000Z', 'sustainability_score': 353, 'carbon_footprint_in_grams': 2460}, {'price': 3310197, 'electricity_price': 3310197, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T12:00:00.000000Z', 'sustainability_score': 347, 'carbon_footprint_in_grams': 2416}, {'price': 3381950, 'electricity_price': 3381950, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T13:00:00.000000Z', 'sustainability_score': 306, 'carbon_footprint_in_grams': 2446}, {'price': 3484194, 'electricity_price': 3484194, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T14:00:00.000000Z', 'sustainability_score': 212, 'carbon_footprint_in_grams': 2308}, {'price': 3644520, 'electricity_price': 3644520, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T15:00:00.000000Z', 'sustainability_score': 111, 'carbon_footprint_in_grams': 2286}, {'price': 4241897, 'electricity_price': 4241897, 'tariff_group': 'high', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T16:00:00.000000Z', 'sustainability_score': 89, 'carbon_footprint_in_grams': 2249}, {'price': 3909993, 'electricity_price': 3909993, 'tariff_group': 'high', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T17:00:00.000000Z', 'sustainability_score': 97, 'carbon_footprint_in_grams': 2230}, {'price': 3708045, 'electricity_price': 3708045, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T18:00:00.000000Z', 'sustainability_score': 134, 'carbon_footprint_in_grams': 2421}, {'price': 3538766, 'electricity_price': 3538766, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T19:00:00.000000Z', 'sustainability_score': 156, 'carbon_footprint_in_grams': 2551}, {'price': 3393687, 'electricity_price': 3393687, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T20:00:00.000000Z', 'sustainability_score': 186, 'carbon_footprint_in_grams': 2418}, {'price': 3304631, 'electricity_price': 3304631, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T21:00:00.000000Z', 'sustainability_score': 218, 'carbon_footprint_in_grams': 2181}, {'price': 3091187, 'electricity_price': 3091187, 'tariff_group': 'normal', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-05T22:00:00.000000Z', 'sustainability_score': 271, 'carbon_footprint_in_grams': 2016}], 'surplus_electricity': 0, 'electricity': 0}}
2023-01-04 21:13:17.545 INFO (MainThread) [custom_components.zonneplan_one.coordinator] _async_update_data: done

2023-01-04 21:13:17.546 DEBUG (MainThread) [custom_components.zonneplan_one.coordinator] Finished fetching zonneplan_one data in 1.127 seconds (success: True)

2023-01-04 21:13:17.659 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Setup sensors for connnection b0e9a247-1a7a-4481-85cf-f35e1c3a6aa1
2023-01-04 21:13:17.659 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan current usage: 0 [0]
2023-01-04 21:13:17.659 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan current usage measured at: None [None]
2023-01-04 21:13:17.659 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan sustainability score: 53.7 [537]
2023-01-04 21:13:17.659 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan current tariff group: normal [normal]
2023-01-04 21:13:17.659 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan current electricity tariff: 0.31201049999999997 [3120105]
2023-01-04 21:13:17.659 WARNING (MainThread) [custom_components.zonneplan_one.coordinator] Could not find measurement_groups of gas_data.measurement_groups.0.meta.price
2023-01-04 21:13:17.660 DEBUG (MainThread) [custom_components.zonneplan_one.coordinator] in {}
2023-01-04 21:13:17.660 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan current gas tariff: None [None]
2023-01-04 21:13:17.660 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan status message:  []
2023-01-04 21:13:17.660 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan status tip: Gemiddeld tarief [Gemiddeld tarief]
2023-01-04 21:13:17.660 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan forcast tariff hour 1: 0.2872902 [2872902]
2023-01-04 21:13:17.660 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan forcast tariff hour 2: 0.2837087 [2837087]
2023-01-04 21:13:17.660 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan forcast tariff hour 3: 0.16198269999999998 [1619827]
2023-01-04 21:13:17.660 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan forcast tariff hour 4: 0.15984099999999998 [1598410]
2023-01-04 21:13:17.660 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan forcast tariff hour 5: 0.16137769999999999 [1613777]
2023-01-04 21:13:17.660 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan forcast tariff hour 6: 0.1554487 [1554487]
2023-01-04 21:13:17.660 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan forcast tariff hour 7: 0.1637614 [1637614]
2023-01-04 21:13:17.660 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan forcast tariff hour 8: 0.1751959 [1751959]
2023-01-04 21:13:17.660 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan forcast tariff group hour 1: low [low]
2023-01-04 21:13:17.660 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan forcast tariff group hour 2: low [low]
2023-01-04 21:13:17.660 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan forcast tariff group hour 3: low [low]
2023-01-04 21:13:17.660 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan forcast tariff group hour 4: low [low]
2023-01-04 21:13:17.660 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan forcast tariff group hour 5: low [low]
2023-01-04 21:13:17.660 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan forcast tariff group hour 6: low [low]
2023-01-04 21:13:17.661 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan forcast tariff group hour 7: low [low]
2023-01-04 21:13:17.661 DEBUG (MainThread) [custom_components.zonneplan_one.sensor] Value Zonneplan forcast tariff group hour 8: low [low]
```

**Comment by ronaldt80** [comment]
I changed the location of the gas price to `summary_data.price_per_hour.24.gas_price` - a bit crude but don't know how to index this otherwise - followed the logic of the current electricity price which takes the 25th available I suspect? This only holds the gas price once per day, eg at 6am. Upon restart it failed loading but assume this means it will find the value at 6am at that time - if you happen to have another suggestion that would be awsome.

I suspect the field you use (measurement_groups of gas_data.measurement_groups.0.meta.price) only becomes available in case the p1 dongle is used; I don't use this yet.

**Comment by ronaldt80** [comment]
Can confirm that with the changes in the code it picks up the right value at 6am, however the sensor is not persistent in case of reboot as it reverts to unknown (which makes sense as by the the value of #24 has changed)

**Comment by fsaris** [comment]
so the `gas_price` is only available at 05:00:00 in the `price_per_hour` listing? I'm assuming you added the `**` around the value.
Can you provide some more examples of `price_per_hour` (few moments during the day).

**Comment by tieskuh** [comment]
> Can confirm that with the changes in the code it picks up the right value at 6am, however the sensor is not persistent in case of reboot as it reverts to unknown (which makes sense as by the the value of array item 24 has changed)

I can confirm that. Wouldn't it be better to update the price every cycle but just check if it is a numeric value >0 before updating. Than the price updates as soon as it is available and not only at 6am.

**Comment by fsaris** [comment]
It's not only updated at 6. But only once a day around 6 (or later when no value is set for that day).

That's why I asked for more examples of `price_per_hour` to find out how to find the value

**Comment by ronaldt80** [comment]
Will share in a moment but structure is identical, except for the gas_price item, so cleanest way would be in my view to poll every iteration, is value is a number then update value, if value is not a number leave the sensor as is.

**Comment by fsaris** [comment]
How does `price_per_hour` look around 10.00 in the morning? Where can we then find the `gas_price` for the current day

**Comment by ronaldt80** [comment]
The gas_price is only there at 6am; making is impossible to look/search the array for a key "gas_price" as it's not there all the time - I noticed though that in your code you already account for a daily update of the gas price . I changed the code to summary.price.24.gas_price - this works, the right price was loaded this morning. Only issue is that upon restart of HA it will revert to unknown (unless it's between 6 and 7 am) - tried to adjust the code but my python skills are too limited I fear.

Will try to solve with an HA automation.

**Comment by fsaris** [comment]
So the rest of the day the gas_price isn't present in data?

If you reinstall the Zonneplan app on your phone during the day is it able to show you the gas price then?

**Comment by ronaldt80** [comment]
It's present in the 5 - 6am GM (6-7 GMT) block, the way you built the code you can actually also see it in the attribute of the electricity price for specifically that time slot. 

The way zonneplan api provides the data seems to always include the past 24hours of data.  So yes, on the phone it is there the whole day.

Cleanest way would in my view be to check hourly if the attribute "gas_price" is available. If so update, if not do nothing. That only leaves the setting of an initial value, conceptually would think that is possible by checking the preceding 6-7 am time slot (eg if it's 6am look at array item 24, if it's 18pm look at item 12 etc, etc)

**Comment by fsaris** [comment]
Ah, the `price_per_hour` includes the past 24hours to. Will check if I can add some extra logic to the coordinator to extract the value this way

**Comment by jperquin** [comment]
As a workaround, I am using the Frank Energie HACS integration (https://github.com/bajansen/home-assistant-frank_energie), which appears quite stable. This produces a reliable sensor: sensor.current_gas_market_price which serves as the basis for the following template sensor:

`- platform: template
  sensors:
    gasprijs_zonneplan:
      friendly_name: Gasprijs Zonneplan
      unit_of_measurement: €/m³
      device_class: monetary
      value_template: >-
        {% set t = now().hour %}
        {% set basis_prijs = state_attr('sensor.current_gas_market_price', 'prices')[t].price | float %}
        {% set var_opslag = 0.0413 %}
        {% set energie_bel = 0.5927 %}
        {% set btw_perc = 0.21 %}
        {{ (basis_prijs + var_opslag + energie_bel) * (1 + btw_perc) }}`

**Comment by ronaldt80** [comment]
Frank is not on EGSI though but TTF EOD, not sure how big the delta is? Alternatively you can also take the value from enever although they also had gas price issues the past few days.

My work around for now is to poll every hour summary.price.24.gas_price, and store it in an input sensor which is persistent when there is a valid numerical value - in practice this is always 6 am in the morning. An automation ensures the input sensor is updated when the price changes.

**Comment by jperquin** [comment]
https://enever.nl/grafieken-gas/

TTF lijkt gemiddeld iets lager danwel bijna gelijk met EGSI.. FRank sensor is dus redelijk goede proxy..

**Comment by dajomas** [comment]
If useful, I have 5 days of debugging information captured and set aside for custom_components.zonneplan_one.api as I would really like to be able to use the gas price in my calculations.

Let me know if it is needed so I can redact it and send it to you

**Comment by ronaldt80** [comment]
Hi Dajomas, 

not ideal but if you replace `gas_data.measurement_groups.0.meta.price` in the python code with `summary.price.24.gas_price` it will pick up the right value once per day. Upon restart however it will loose the value so I store it in an input_number through an automation. See below, only thing you need to do is create a helper value/input_number and adjust the automation to reflect the name you selected for this input_number.

It's a bit crude but effective!

```
alias: Update gas price
description: ""
trigger:
  - platform: state
    entity_id:
      - sensor.zonneplan_current_gas_tariff
    from: unknown
  - platform: state
    entity_id:
      - sensor.zonneplan_current_gas_tariff
    from: unavailable
  - platform: state
    entity_id:
      - sensor.zonneplan_current_gas_tariff
  - platform: time
    at: "06:00:30"
condition:
  - condition: numeric_state
    entity_id: sensor.zonneplan_current_gas_tariff
    above: 0.01
action:
  - service: input_number.set_value
    data:
      value: "{{states.sensor.zonneplan_current_gas_tariff.state}}"
      entity_id: input_number.gas
mode: queued
max: 10
```

**Comment by dajomas** [comment]
Thanx @ronaldt80 I will give it a shot. But a more permanent solution would be very welcome...

**Comment by Inferi0r** [comment]
> Ah, the `price_per_hour` includes the past 24hours to. Will check if I can add some extra logic to the coordinator to extract the value this way

I can confirm that gas price is unavailable too after my contract started.

> Hi Dajomas,
> 
> not ideal but if you replace `gas_data.measurement_groups.0.meta.price` in the python code with `summary.price.24.gas_price` it will pick up the right value once per day. Upon restart however it will loose the value so I store it in an input_number through an automation. See below, only thing you need to do is create a helper value/input_number and adjust the automation to reflect the name you selected for this input_number.
> 
> It's a bit crude but effective!
> 
> ```
> alias: Update gas price
> description: ""
> trigger:
>   - platform: state
>     entity_id:
>       - sensor.zonneplan_current_gas_tariff
>     from: unknown
>   - platform: state
>     entity_id:
>       - sensor.zonneplan_current_gas_tariff
>     from: unavailable
>   - platform: state
>     entity_id:
>       - sensor.zonneplan_current_gas_tariff
>   - platform: time
>     at: "06:00:30"
> condition:
>   - condition: numeric_state
>     entity_id: sensor.zonneplan_current_gas_tariff
>     above: 0.01
> action:
>   - service: input_number.set_value
>     data:
>       value: "{{states.sensor.zonneplan_current_gas_tariff.state}}"
>       entity_id: input_number.gas
> mode: queued
> max: 10
> ```

Hi ronald, thanks for your help. Maybe good to mention that the value in the python code is within the file: /config/custom_components/zonneplan_one/const.py

When creating input helper:
- Name 'gas' automatically becomes entity 'input_number.gas'
- Unit of measurement: €/m³ (like original Zonneplan sensor)
- Step size: 0,000001 (fits with gas prices as reported by Enever)
- Min-Max: 0-4 (or typically other euro range)
- Icon: mdi:cash (original sensor but i like mdi:gas-burner too)

**Comment by dajomas** [comment]
Unfortunately, it did not work.

In fact, I don't have a gas price nor an electricity price in Home Assistant from the Zonneplan One custom component

> Hi Dajomas,
> 
> not ideal but if you replace `gas_data.measurement_groups.0.meta.price` in the python code with `summary.price.24.gas_price` it will pick up the right value once per day. Upon restart however it will loose the value so I store it in an input_number through an automation. See below, only thing you need to do is create a helper value/input_number and adjust the automation to reflect the name you selected for this input_number.
> 
> It's a bit crude but effective!
> 
> ```
> alias: Update gas price
> description: ""
> trigger:
>   - platform: state
>     entity_id:
>       - sensor.zonneplan_current_gas_tariff
>     from: unknown
>   - platform: state
>     entity_id:
>       - sensor.zonneplan_current_gas_tariff
>     from: unavailable
>   - platform: state
>     entity_id:
>       - sensor.zonneplan_current_gas_tariff
>   - platform: time
>     at: "06:00:30"
> condition:
>   - condition: numeric_state
>     entity_id: sensor.zonneplan_current_gas_tariff
>     above: 0.01
> action:
>   - service: input_number.set_value
>     data:
>       value: "{{states.sensor.zonneplan_current_gas_tariff.state}}"
>       entity_id: input_number.gas
> mode: queued
> max: 10
> ```

**Comment by fsaris** [comment]
> Unfortunately, it did not work.
> 
> In fact, I don't have a gas price nor an electricity price in Home Assistant from the Zonneplan One custom component
> 

What devices do you see in Home assistant for this integration?

**Comment by ronaldt80** [comment]
@dajomas - can you share some of this earlier debug output? Did solar pricing work before you made the changes to the code? If so, can you share some more recent log output?

**Comment by dajomas** [comment]
> 
> What devices do you see in Home assistant for this integration?

I have "Connect energiemeter", "PV Installatie" and 2 times "Usage" (both are for electricity)

**Comment by dajomas** [comment]
> @dajomas - can you share some of this earlier debug output? Did solar pricing work before you made the changes to the code? If so, can you share some more recent log output?

I have a zonneplan contract since Jan 6, 2023 and I have not seen any pricing information in Home Assistant for either electricity, nor gas. Attached is the latest (redacted) debug output
[debug.redact.txt](https://github.com/fsaris/home-assistant-zonneplan-one/files/10454736/debug.redact.txt)

**Comment by fsaris** [comment]
Prices are only available when you have a energy contract with Zonneplan.

**Comment by dajomas** [comment]
> Prices are only available when you have a energy contract with Zonneplan.

I have a contract since Jan 6, 2023

**Comment by ronaldt80** [comment]
Maybe try to remove the integration, delete the entities and reinstall the integration?

**Comment by dajomas** [comment]
> Maybe try to remove the integration, delete the entities and reinstall the integration?

Alas, that did not work. The same 4 devices and 19 entities came back after I made sure they were gone after I deleted the integration

**Comment by Inferi0r** [comment]
@ronaldt80 my automation did not pick-up anything too, tonight, however I do see these records in my log:

'tariff_group': 'low', 'solar_percentage': 0, 'solar_yield': 0, 'datetime': '2023-01-19T04:00:00.000000Z', 'sustainability_score': 328, 'carbon_footprint_in_grams': 2312}, {'price': 2993661, 'electricity_price': 2993661, **'gas_price': 13753959**

**Comment by ronaldt80** [comment]
Yes, it is indeed so extracting it from the attribute is also an option.

Still odd though - mine picked it up at exactly 6am and subsequently triggered the automation. Will share my repo tonight. Perhaps it's just a minor difference in the code.

**Comment by ronaldt80** [comment]
> > Maybe try to remove the integration, delete the entities and reinstall the integration?
> 
> Alas, that did not work. The same 4 devices and 19 entities came back after I made sure they were gone after I deleted the integration

The only thing I can think of is to check your debug log for today. In addition, just to double check, do you have the price info in your zonneplan app?

**Comment by Inferi0r** [comment]
Discussing with relatives who also joined Zonneplan recently and do use this package: I think the issue may have to do with not having the Zonneplan official P1 meter plugged in, I will confirm this week by plugging in. 

However, for now I obtained my gasprices with help of this in _/config/sensor.yaml_ I think this is the most easy solution as no helpers nor automations are required.

- platform: template
  sensors:
    zonneplan_current_gas_tarriff_fixed: 
      friendly_name: "Gas price Zonneplan"
      unit_of_measurement: €/m³
      value_template: >
        {{ (state_attr('sensor.zonneplan_current_electricity_tariff', 'forcast')|regex_findall_index("'gas_price':\s(\d{8})"))|float|round(0)/10000000 }}

**Comment by dajomas** [comment]
I checked the priceinfo in my Zonneplan app and I can see all the prices back to January 1st. 

I also have the P1 meter plugged in since half december (using a splitter provided by Zonneplan, because my HA instance is also reading the P1 data)

**Comment by dajomas** [comment]
> Discussing with relatives who also joined Zonneplan recently and do use this package: I think the issue may have to do with not having the Zonneplan official P1 meter plugged in, I will confirm this week by plugging in. I am still awaiting the Zonneplan splitter which are out of stock now for 2-3 weeks, but can do a swap with my own P1.
> 
> However, for now I obtained my gasprices with help of this new sensor in my _/config/sensor.yaml_ I believe this is the most easy solution as no helpers nor automations are required and the solution is not dependent on HA restarts.
> 
> ```
> - platform: template
>   sensors:
>     zonneplan_current_gas_tarriff_fixed: 
>       friendly_name: "Gas price Zonneplan"
>       unit_of_measurement: €/m³
>       value_template: >
>         {{ (state_attr('sensor.zonneplan_current_electricity_tariff', 'forcast')|regex_findall_index("'gas_price':\s(\d{8})"))|float|round(0)/10000000 }}
> ```

That COULD work, if only the sensor.zonneplan_current_electricity_tariff was available.... Unfortunately, my integration doesn't provide me any tariffs

**Comment by ronaldt80** [comment]
Can you share a debug log for today?

**Comment by dajomas** [comment]
> Can you share a debug log for today?

Yep, will extract and redact today's zonneplan debug output and upload it when it's ready

**Comment by dajomas** [comment]
> Can you share a debug log for today?

Here it is
[zonneplan.2023-01-20.redact.log](https://github.com/fsaris/home-assistant-zonneplan-one/files/10463974/zonneplan.2023-01-20.redact.log)

**Comment by ronaldt80** [comment]
Ok, so the good news is that the prices are there. Even the gas price. Will have a further look tonight.

Maybe a stupid/obvious question - electricity price is disabled by default, did you enable it in the integration? Just checking..

**Comment by dajomas** [comment]
> Ok, so the good news is that the prices are there. Even the gas price. Will have a further look tonight.
> 
> Maybe a stupid/obvious question - electricity price is disabled by default, did you enable it in the integration? Just checking..

Nope, the only change I made is the one you suggested earlier

**Comment by ronaldt80** [comment]
Ok.. but if the sensor is not enabled in the integration then you will not find it either. Please try by going to the integration, device "usage" click on "entities not shown" and enable the ones related to tariff.

**Comment by dajomas** [comment]
Yep, that did the trick. Never needed to enable entities so didn't know that was an option. Thanx

**Comment by RobReus** [comment]
Just putting this out here as it might help troubleshooting, the gas prices were also not showing for me even though they did appear once a day (6AM iirc) in the prices from the `/summary` API endpoint. The moment I connected my Zonneplan One logger to my P1 port, the gas prices started showing in Home Assistant.

**Comment by ronaldt80** [comment]
I also have the P1 meter connected. Just installed the new version of this integration, still no gas price - just a few more errors (Could not find charge_point of charge_point) - will open a new issue for this but @fsaris if you have any thoughts as to how to take the gas price from the sensor that always works (see above) that would be great.

**Comment by fsaris** [comment]
> Could not find charge_point of charge_point
You are using the Beta?

This message should only show up once when the integration is started and be logged as a warning.

Will dive into the gas price issue today

**Comment by hmmbob** [comment]
Still strange that it doesn't show for some... Works fine here. 

![Screenshot_20230212_103242_Home Assistant](https://user-images.githubusercontent.com/33529490/218303294-92136168-b794-49e6-afa0-f16692ee0058.jpg)

You did enable the gas tariff sensor, right?

**Comment by ronaldt80** [comment]
Well aware, and strange indeed - it's almost as if different users get different data sets from the api.

 In the first post I included the background, the field which is used to fill the sensor data (gas_data.measurement_groups.0.meta.price) is not available in my data set. I do have gas price data in the price attribute. If you have suggestions to cater for this that be much appreciated.

**Comment by fsaris** [comment]
@ronaldt80 can you try https://github.com/fsaris/home-assistant-zonneplan-one/releases/tag/0.1.17b5

**Comment by ronaldt80** [comment]
Just installed, first iteration was successful, notice you reference summary_data.gas_price directly -didnt know this was possible! Thanks a lot- will keep an eye on logs tomorrow and let you know

**Comment by fsaris** [comment]
>  notice you reference summary_data.gas_price directly

the value is fetched from the hourly_prices list

**Comment by Inferi0r** [comment]
Gas price is finally working for me on Beta 5, immediately after reboot.

**Comment by ronaldt80** [comment]
No issues whatsoever with gas price since the last changes were incorporated. Many thanks @fsaris !

**Comment by fsaris** [comment]
It's not in a stable release yet so let's keep this ticket open until it is.

**Comment by fsaris** [comment]
fix is released
